sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierTest.cs

```diff
@@ -0,0 +1,80 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+
+namespace SonarAnalyzer.UnitTest.TestFramework
+{
+    [TestClass]
+    public class VerifierTest
+    {
+        [TestMethod]
+        public void VerifyCodeFix_WithDuplicateIssues()
+        {
+            const string filename = @"TestCases\VerifyCodeFix.Empty.cs";
+            System.Action a = () => { Verifier.VerifyCodeFix(filename, filename, new TestDuplicateLocationRule(), new TestDuplicateLocationRuleCodeFix()); };
+            a.Should().NotThrow();
+        }
+
+        [Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer(LanguageNames.CSharp)]
+        private class TestDuplicateLocationRule : SonarDiagnosticAnalyzer
+        {
+            public const string DiagnosticId = "Test42";
+            private DiagnosticDescriptor rule = new DiagnosticDescriptor(DiagnosticId, "Title", "Message", "Category", DiagnosticSeverity.Warning, true, null, null, new[] { "MainSourceScope" });
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+            protected override void Initialize(SonarAnalysisContext context)
+            {
+                context.RegisterSyntaxNodeAction<SyntaxKind>(c =>
+                {
+                    // Duplicate issues from different analyzer versions, see https://github.com/SonarSource/sonar-dotnet/issues/1109
+                    c.ReportDiagnostic(Diagnostic.Create(this.rule, c.Node.GetLocation()));
+                    c.ReportDiagnostic(Diagnostic.Create(this.rule, c.Node.GetLocation()));
+                }, SyntaxKind.NamespaceDeclaration);
+            }
+        }
+
+        private class TestDuplicateLocationRuleCodeFix : SonarCodeFixProvider
+        {
+            public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(TestDuplicateLocationRule.DiagnosticId);
+            public override FixAllProvider GetFixAllProvider() => DocumentBasedFixAllProvider.Instance;
+
+            protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
+            {
+                // Do not change the document
```

Comment:
I would remove the comment, it doesn't provide a lot of info.

Reply:
Ok. It seemed weird to "do nothing" so I wanted to comment that it's on purpose.